src/main/java/com/gmail/inverseconduit/bot/AbstractBot.java

```diff
@@ -4,24 +4,59 @@
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.LinkedBlockingQueue;
-import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
 import com.gmail.inverseconduit.chat.ChatWorker;
 import com.gmail.inverseconduit.datatype.ChatMessage;
 
+/**
+ * Abstract Class that implements default ChatWorker behavior.
+ * Internally this class exposes the protected fields: {@link #messageQueue} and
+ * {@link #processingThread}. </br>
+ * This class implements thread-safe behavior to
+ * enqueue messages to {@link #messageQueue}.
+ * </br></br>If the {@link ChatWorker#POISON_PILL POISON_PILL} is enqueued,
+ * {@link #shutdown()} will be called to notify implementing classes of the
+ * shutdown request.
+ * Also the {@link #processingThread}'s {@link ExecutorService#shutdown()
+ * shutdown}-method will be called. It is accordingly not usable after this.
+ * 
+ * @author Vogel612<<a href="mailto:vogel612@gmx.de"
+ *         >vogel612@gmx.de</a>>
+ */
```

Comment:
That Javadoc format sounds interesting. Currently I just ran the shared formatting preferences, maybe they can be adapted to include this javadoc format. Thoughts @Unihedro ?